### PR TITLE
Add support for passing launch text for Cap projects to generate README files

### PR DESCRIPTION
**Pull request by kjose90**
https://github.com/SAP/open-ux-tools/issues/2019

Add support to dynamically generate launch text for Cap projects in `cap-config-writers`

### CODE REVIEW

**Review by devinea — DISMISSED** [review]
Code changes and tests look good.
Patchset ✅

File: packages/cap-config-writer/src/translations/cap-config-writer.i18n.json [hunk 1]
```diff
@@ -8,5 +8,6 @@
     },
     "error": {
         "updateApplicationYaml": "Error occured while updating application yaml {{ error }}"
-    }
+    },
+    "TEXT_LAUNCH_CAP": "In order to launch the generated app, simply start your CAP project{{mvnCommand}} and navigate to the following location in your browser:\n\n{{- capUrl}}"
```
Comment (IainSAP) [inline packages/cap-config-writer/src/translations/cap-config-writer.i18n.json:12]: Key casing?

File: packages/cap-config-writer/src/cap-writer/utils.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        } else {
+            // For Node.js projects or when capType is undefined
+            return `http://localhost:4004/${project}/index.html`;
+        }
+    }
+
+    let mvnCommand = '';
+    if (capType === 'Java') {
+        // If capType is 'Java', set the Maven command for running Spring Boot
+        mvnCommand = ' (```mvn spring-boot:run```)';
```
[the remark this answers is not in the document]
Comment (IainSAP) [reply packages/cap-config-writer/src/cap-writer/utils.ts:31]: I agree, can we re-implement this function please @kjose90 ?

File: packages/cap-config-writer/src/cap-writer/utils.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * Generates a launch text for the launching of applications.
+ *
+ * @param capType The type of CAP runtime used in the application will be provided for CAP applications.
+ * @param projectName The project's name, which is the module name.
+ * @param appId If appId is provided, it will be used to open the application instead of the project name. This option is available for use with npm workspaces.
+ * @returns The launch text for the application.
+ */
+export function getLaunchText(capType: CapRuntime, projectName: string, appId?: string): string {
```
Comment (IainSAP) [inline packages/cap-config-writer/src/cap-writer/utils.ts:31]:
```suggestion
export function getAppLaunchText(capType: CapRuntime, projectName: string, appId?: string): string {
```

File: packages/cap-config-writer/src/cap-writer/utils.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { t } from '../i18n';
+import type { CapRuntime } from '@sap-ux/odata-service-inquirer';
+
+/**
+ * Determines the URL based on the project type.
```
Comment (IainSAP) [inline packages/cap-config-writer/src/cap-writer/utils.ts:5]:
```suggestion
 * Returns the url to the specified cap app as served by `cds serve` or `cds watch`
```

**Review by IainSAP — COMMENTED** [review]
Minor naming/comments comments

**Review by IainSAP — APPROVED** [review]
Thanks @kjose90 

-  Tests added to cover new utility functions

**Review by devinea — APPROVED** [review]
Code changes are clear and covered by tests
changeset - ok, patch would probably suffice here.
Thanks

**Comment by kjose90** [comment]
Thanks every one for reviewing

**Pull request merged**